lib/ansible/modules/packaging/language/pip.py

```diff
@@ -234,6 +252,53 @@
 _SPECIAL_PACKAGE_CHECKERS = {'setuptools': 'import setuptools; print(setuptools.__version__)',
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
+_VCS_RE = re.compile(r'(svn|git|hg|bzr)\+')
+
+op_dict = {">=": operator.ge, "<=": operator.le, ">": operator.gt,
+           "<": operator.lt, "==": operator.eq, "!=": operator.ne, "~=": operator.ge}
+
+
+def _is_vcs_url(name):
```

Comment:
Remark: with regexp factored out this could become just
```python
from functools import partial
...
_is_vcs_url = partial(re.match, _VCS_RE)
"""Test whether a name is package name or version specifier."""
```

(But I bet Toshio would blame it for being unreadable, so don't do it :smile:)

Reply:
I wouldn't write a function which was itself a single function call ;-)